### ISSUE TITLE
dataSourceErrorOccured

### ISSUE THREAD

**Issue by KenmuHuang** [tumtumtum/StreamingKit#398]
I can not do anything when I play an empty audio from server.
It never execute public delegate method in the iOS 11.

The console show related information as following:
**attemptReconnect 0/0
dataSourceErrorOccured
attemptReconnect 0/0
dataSourceErrorOccured
...**

For example I access this audio:
http://www.tingban.cn/record/file_play/Mg/NDUxOV8xNDMwXzY1Njg0XzI/L3dpenphcmRhdWRpbzEvMjAxNDA0LzJlMzJmNzEyLWZjMGYtNDk3NC1iYTlkLTNiYTNlNmRhZjdmMS5tcDM.mp3

**Comment by patrickjquinn** [comment]
I'm getting a 404 when I hit that link. Are you testing against dead links?

**Comment by KenmuHuang** [comment]
@patrickjquinn 
Yes, I want to get a notice when I play invalid audio link. If I getting a notice(e.g. delegate), I can do somethig(e.g. play next audio) for it.

**Comment by xiaogehenjimo** [comment]
@KenmuHuang 大佬 传了无效的地址 控制台一直打印这个 请问在哪里能获得这个事件?

**Comment by KenmuHuang** [comment]
> @KenmuHuang 大佬 传了无效的地址 控制台一直打印这个 请问在哪里能获得这个事件?

尴尬，之前就是因为同样情况获取不到，所以提问；现在不知道有没新增功能支持这处理

**Comment by diegostamigni** [comment]
Hi @KenmuHuang 

> It never execute public delegate method in the iOS 11
I'm not sure I understand this. Do you mean that no method in protocol STKAudioPlayerDelegate gets called?

**Comment by youyou5920** [comment]
let audioDataSource = STKAudioPlayer.dataSource(from: sourceURL)

self.audioPlayer = STKAudioPlayer(options: STKAudioPlayerOptions())
self.audioPlayer?.delegate = self

self.audioPlayer?.play(audioDataSource)

// If sourceURL is invalid 404, it will trigger dataSourceErrorOccured in a loop,
// Do not execute : -(void) audioPlayer:(STKAudioPlayer*)audioPlayer unexpectedError:(STKAudioPlayerErrorCode)errorCode

**Comment by iDevelopper** [comment]
Hi there,

Any news about this? No public delegate method is called after dataSourceErrorOccured...

Thank you.

**Comment by MQL9011** [comment]
I have the same bug.

Play url error , but delegate is not called. 

Is anyone solve this?

**Comment by iDevelopper** [comment]
I had to absolutely have a response in my audio palyer delegate for this error. So I wrote this category which I hope will help some of you. It calls the method audioPlayer:unexpectedError with STKAudioPlayerErrorDataSource error.

[Archive.zip](https://github.com/tumtumtum/StreamingKit/files/3287263/Archive.zip)
